src/main/java/de/tum/in/www1/artemis/service/connectors/athena/AthenaModuleService.java

```diff
@@ -78,35 +79,24 @@ private List<AthenaModuleDTO> getAthenaModules() throws NetworkingException {
     }
 
     /**
-     * Get all available Athena programming modules for a specific course.
+     * Get all available Athena modules for a specific course and exercise type
      *
-     * @param course The course for which the modules should be retrieved
+     * @param course       The course for which the modules should be retrieved
+     * @param exerciseType The exercise type for which the modules should be retrieved
      * @return The list of available Athena text modules for the course
      * @throws NetworkingException is thrown in case the modules can't be fetched from Athena
      */
-    public List<String> getAthenaProgrammingModulesForCourse(Course course) throws NetworkingException {
-        List<String> availableProgrammingModules = getAthenaModules().stream().filter(module -> "programming".equals(module.type)).map(module -> module.name).toList();
-        if (!course.getRestrictedAthenaModulesAccess()) {
-            // filter out restricted modules
-            availableProgrammingModules = availableProgrammingModules.stream().filter(moduleName -> !restrictedModules.contains(moduleName)).toList();
-        }
-        return availableProgrammingModules;
-    }
+    public List<String> getAthenaModulesForCourse(Course course, ExerciseType exerciseType) throws NetworkingException {
+
+        final String exerciseTypeName = exerciseType.getExerciseTypeAsReadableString();
+
+        List<String> availableModules = getAthenaModules().stream().filter(module -> exerciseTypeName.equals(module.type)).map(module -> module.name).toList();
 
-    /**
-     * Get all available Athena text modules for a specific course.
-     *
-     * @param course The course for which the modules should be retrieved
-     * @return The list of available Athena text modules for the course
-     * @throws NetworkingException is thrown in case the modules can't be fetched from Athena
-     */
-    public List<String> getAthenaTextModulesForCourse(Course course) throws NetworkingException {
-        List<String> availableTextModules = getAthenaModules().stream().filter(module -> "text".equals(module.type)).map(module -> module.name).toList();
         if (!course.getRestrictedAthenaModulesAccess()) {
             // filter out restricted modules
-            availableTextModules = availableTextModules.stream().filter(moduleName -> !restrictedModules.contains(moduleName)).toList();
+            availableModules = availableModules.stream().filter(moduleName -> !restrictedModules.contains(moduleName)).toList();
         }
-        return availableTextModules;
+        return availableModules;
```

Comment:
```suggestion
         Stream<String> availableModules = getAthenaModules().stream().filter(module -> exerciseTypeName.equals(module.type)).map(module -> module.name);
         if (!course.getRestrictedAthenaModulesAccess()) {
            // filter out restricted modules
            availableModules = availableModules.filter(moduleName -> !restrictedModules.contains(moduleName)).toList();
        }
        return availableModules.toList();
```
This suggestion sadly only references the last line but should reference beginning from line 93